scylla/src/transport/topology.rs

```diff
@@ -1602,6 +1615,28 @@ fn parse_cql_type(p: ParserState<'_>) -> ParseResult<(PreCqlType, ParserState<'_
         })?;
 
         Ok((PreCqlType::Tuple(types), p))
+    } else if let Ok(p) = p.accept("vector<") {
+        let (inner_type, p) = parse_cql_type(p)?;
+
+        let p = p.skip_white();
+        let p = p.accept(",")?;
+        let p = p.skip_white();
+
+        let (size, p) = p.take_while(|c| c.is_numeric());
+        let size = size.parse().map_err(|_| {
+            p.error(ParseErrorCause::Other(
+                "Expected integer but found non-numeric character",
+            ))
+        })?;
```

Comment:
The error message is slightly misleading. After all, `4.0` is a valid number, so `.` is a numeric character, but it's never a part of an integer.